<launch>
  <node name="map_server" pkg="map_server" type="map_server" args="$(find adversarial_coverage)/maps/lizi_lab_straight.pgm 0.05" respawn="false"/>
    
  <node pkg="laser_scan_matcher" type="laser_scan_matcher_node" 
    name="laser_scan_matcher_node" output="screen" respawn="true"> 
    <param name="fixed_frame" value="lizi_2_odom"/>
    <param name="base_frame" value="lizi_2_base_link"/>
    <param name="use_cloud_input" value="false"/>
    <param name="publish_tf" value="true"/>
    <param name="publish_odom" value="false"/>
    <param name="use_imu" value="false"/>
    <param name="use_vel" value="false"/>
    <param name="use_odom" value="false"/>
    <param name="kf_dist_linear" value="-1"/>

    <param name="max_iterations" value="13"/>
    <param name="max_correspondence_dist" value="0.8"/>
    <param name="epsilon_xy" value="0.0001"/>

    <remap from="scan" to="/lizi_2/scan" />
    <remap from="odom" to="/lizi_2/odom_pub" /> 
  </node>
  
  <node pkg="amcl" type="amcl" name="amcl" respawn="true">
    <param name="initial_pose_x" value="0.69375" />
    <param name="initial_pose_y" value="0.69375" />
    <param name="initial_pose_a" value="0" />

    <param name="odom_model_type" value="omni"/>
    <param name="transform_tolerance" value="0.3" />
    <param name="gui_publish_rate" value="0.1"/>
    <param name="laser_max_beams" value="60"/>
    <param name="min_particles" value="250"/>
    <param name="kld_err" value="0.01"/>
    <param name="kld_z" value="0.99"/>
    <param name="odom_alpha1" value="0.2"/>
    <param name="odom_alpha2" value="0.2"/>
    <param name="odom_alpha3" value="0.8"/>
    <param name="odom_alpha4" value="0.2"/>

    <param name="odom_alpha5" value="0.2"/>
    <param name="laser_z_hit" value="0.5"/>
    <param name="laser_z_short1" value="0.05"/>
    <param name="laser_z_max" value="1.05"/>
    <param name="laser_z_rand" value="0.5"/>
    <param name="laser_sigma_hit" value="0.2"/>
    <param name="laser_lambda_short" value="0.1"/>
    <param name="laser_lambda_short" value="0.1"/>
    <param name="laser_model_type" value="likelihood_field"/>
    <param name="laser_likelihood_max_dist" value="2.0"/>
    <!--<param name="update_min_d" value="0.2"/>-->
    <param name="update_min_d" value="0.01"/>
    <!--<param name="update_min_a" value="0.3"/>-->
    <param name="update_min_a" value="0.005"/>    
    <param name="resample_interval" value="1"/>
    <param name="transform_tolerance" value="0.1"/>
    <param name="recovery_alpha_slow" value="0.0"/>
    <param name="recovery_alpha_fast" value="0.0"/>
    <remap from="scan" to="/lizi_2/scan" />
    <remap from="map" to="/map" />
    <param name="map_frame_id" value="map"/>
    <param name="odom_frame_id" value="lizi_2_odom"/>
    <param name="base_frame_id" value="lizi_2_base_link"/>
 -a    <param name="max_particles" value="1000"/>
   </node>

  <param name="robot_width" value="0.4625" />
  <param name="robot_length" value="0.4625" />
  <!--<param name="high_linear_speed" value="1.0" />-->
  <param name="high_linear_speed" value="0.6" />
  <!--<param name="low_linear_speed" value="0.5" />-->
  <param name="low_linear_speed" value="0.6" />
  <!--<param name="angular_speed" value="1.0" />-->
  <param name="angular_speed" value="0.5" />
  <!--<param name="linear_tolerance" value="0.015" />-->
  <param name="linear_tolerance" value="0.05" />
  <!--<param name="angular_tolerance" value="0.0015" />-->
  <param name="angular_tolerance" value="0.0085" />
  <param name="rotation_fix_tolerance" value="0.1025" />
  <param name="rotation_fix_speed" value="0.5" />
  <param name="risk_factor" value="1" />
  <param name="threats_file" value="$(find adversarial_coverage)/maps/lizi_empty_threats.inc"/>
  <param name="log_file" value="$(find adversarial_coverage)/output/lizi_adversarial_coverage.txt" />
  <param name="write_log" value="true" />
  <param name="stat_file" value="$(find adversarial_coverage)/output/lizi_adversarial_coverage_stat" />
  <param name="robot_name" value="lizi2" />
  <param name="simulation_mode" value="false" /> 

  <!-- Run coverage node -->
  <node name="adversarial_coverage" pkg="adversarial_coverage" type="adversarial_coverage_node" output="screen" cwd="node">
    <remap from="cmd_vel" to="/lizi_2/cmd_vel" />
  </node>
</launch>
